armi/physics/neutronics/fissionProductModel/fissionProductModel.py

```diff
@@ -185,7 +192,25 @@ def setAllComponentFissionProducts(self):
         """
         for b in self.r.core.getBlocks(includeAll=True):
             b.setLumpedFissionProducts(None)
-            for c in b.getComponents(Flags.DEPLETABLE):
+
+            # If detailed axial expansion is active, mapping between blocks occurs on uniform mesh
+            # and this can cause blocks to have isotopes that they dont have cross sections for/
+            # Fix this by adding all isotopes to all blocks so they are present it lattice physics.
+            allBlocksNeedAllNucs = self.cs[CONF_DETAILED_AXIAL_EXPANSION]
+
+            compsToAddIso = b.getComponents(Flags.DEPLETABLE)
+            if allBlocksNeedAllNucs and not compsToAddIso:
+                # add the isotopics to the smallest solid since that is usually the most "interesting"
+                # sorted() calls getBoundingCircleOuterDiameter under the hood
+                solidsOrderedBySize = sorted(c for c in b if c.containsSolidMaterial())
+                if solidsOrderedBySize:
+                    compsToAddIso = [solidsOrderedBySize[0]]
+                else:
+                    # no solids, so just add to smallest component
+                    compsToAddIso = [
+                        sorted(c for c in b if c.containsSolidMaterial())[0]
```

Comment:
This line seems to produce the same exact `compsToAddIso` as the one in the `if` part of the block. Should `if c.containsSolidMaterial()` be removed here?

Reply:
yes! thanks!